File: src/main/java/com/xelita/commons/exception/NotFoundException.java
package com.xelita.commons.exception;

import com.xelita.commons.Identifiable;
import lombok.Getter;
import org.springframework.http.HttpStatus;

/**
 * NotFoundException.
 *
 * @author xelita
 */
public class NotFoundException extends TrackableException {

    private static final String EXCEPTION_KEY = "NOT_FOUND_EXCEPTION";


    public NotFoundException(Identifiable identifiable) {
        super(identifiable.getType() + " with id [" + identifiable.getId() + "] could not be found,", EXCEPTION_KEY, HttpStatus.NOT_FOUND.value());
    }

    public NotFoundException(Identifiable identifiable, Throwable th, int statusCode) {
        super(identifiable.getType() + " with id [" + identifiable.getId() + "] could not be found,", th, EXCEPTION_KEY, HttpStatus.NOT_FOUND.value());
    }
}

File: src/main/kotlin/com/xelita/commons/exception/NotFoundException.kt
package com.xelita.commons.exception

import com.xelita.commons.Identifiable
import org.springframework.http.HttpStatus

/**
 * NotFoundException.
 *
 * @author xelita
 */
class NotFoundException : TrackableException {

    constructor(obj: Identifiable) : super(obj.getType() + " with id [" + obj.getId() + "] could not be found,", "NOT_FOUND_EXCEPTION", HttpStatus.NOT_FOUND.value())

    constructor(obj: Identifiable, th: Throwable) : super(obj.getType() + " with id [" + obj.getId() + "] could not be found,", th, "NOT_FOUND_EXCEPTION", HttpStatus.NOT_FOUND.value())
}